src/pages/register-page.tsx

```diff
@@ -37,17 +50,153 @@ const styles = StyleSheet.create({
+                // First we check if this this devices has already logged in before
+                AsyncStorage.getItem(userAuthIDAsyncStorageName).then(data => {
+                    if (data) {
+                        startLogin(data);
+                    }
+                    else {
+                        setLoginLoading(false);
+                    }
+                })
+            }
+          });
+      
+          return () => task.cancel();
+        }, [])
+      );
+
+    const checkPhoneNumber = () => {
+        setPhoneNumber(inputValue)
+        if (inputValue.length === numberLength) {
+            sendVerificationCode(inputValue);
+        }
+        else {
+            setError(true);
+        }
+    }
 
-    const checkPhoneNumber = () => {              //placeholder for actual function when using phone number to identify user
-        if (numberInput.length === 10) {
-            setPhoneNumber(numberInput)
-            navigation.navigate('Home');
+    const sendVerificationCode = async (number: string) => {
+        if (recaptchaVerifier.current) {
+            const phoneProvider = new firebase.auth.PhoneAuthProvider();
+            setInputValue('');
+            const verificationId = await phoneProvider.verifyPhoneNumber(
+                    inputValue,
+                    recaptchaVerifier.current
+            );
+            setVerificationID(verificationId);
         }
-        setError(numberInput.length !== 10)
+    }
+
+    const requestCredentials = async (verificationCode: string, verificationID: string) => {
+        setLoginLoading(true);
+        try {
+            // Try to verify the credentials
+            const credential = firebase.auth.PhoneAuthProvider.credential(
+                verificationID,
+                verificationCode
+            );
+
+            // Get a auth user
+            const userAuth = await firebase.auth().signInWithCredential(credential);
+            const userID = userAuth.user?.uid;
+
+            // Saving the user and continue to login
+            if (userID) {
+                await AsyncStorage.setItem(userAuthIDAsyncStorageName, userID);
+                await startLogin(userAuthIDAsyncStorageName);
+                setAuthenticationError(false);
+            }
+        } catch (err) {
+            console.log(err);
+            setAuthenticationError(true);
+        }
+        setLoginLoading(false);
+    }
+
+    const startLogin = async (userAuthID: string) => {
+
+        // First we need to check if this user exist
+        const user = await UserModel.getUserByAuth(userAuthID);
+        console.log(user);
+        if (user) {
+            props.setUserData(user);
+            props.router.navigation.replace('Home');
+            return;
+        }
+
+        // No user found, need to create a new one
+        const newUser: User = {
+            authID: userAuthID,
+            address: null,
+            name: null,
+            phone: null,
```

Comment:
```suggestion
            phone: phoneNumber,
```
Couldn't we add the phone number directly here?

Reply:
That true!